File: TodoAuthApi/Models/Todo.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TodoAuthApi.Models
{
    public class Todo
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(maximumLength: 200)]
        public String Title { get; set; } = String.Empty;
        public String? Description { get; set; }
        public bool IsCompleted { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int UserId { get; set; }
        [ForeignKey(nameof(UserId))]
        public User User { get; set; } = null!;
    }
}

File: TodoFrontend/src/types/todo.ts
export interface TodoResponse{
    id: number;
    title: string;
    description?: string;
    isCompleted: boolean;
    createdAt: string; // APIではDateTime型で扱っているが、フロントエンドでは文字列として受け取るためstring型にしている
    updatedAt: string;
}

export interface TodoCreateRequest{
    title: string;
    description?: string;
}

export interface TodoUpdateRequest{
    title :string;
    description?: string;
    isCompleted: boolean;
}